### PR TITLE
Clear entityInsertions for specific entityName

### PR DESCRIPTION
**Pull request by ricohumme**
See https://github.com/doctrine/doctrine2/issues/5849

### CODE REVIEW

**Comment by Ocramius** [comment]
Requires test cases

**Comment by ricohumme** [comment]
Test case provided

File: lib/Doctrine/ORM/UnitOfWork.php [hunk 1]
```diff
@@ -2401,6 +2401,13 @@ public function clear($entityName = null)
                     $this->doDetach($entity, $visited, false);
                 }
             }
+
+            foreach ($this->entityInsertions as $hash => $entity) {
+                if (get_class($entity) != $entityName) {
```
Comment (Ocramius) [inline lib/Doctrine/ORM/UnitOfWork.php]: Strict comparison needed here

File: lib/Doctrine/ORM/UnitOfWork.php [hunk 2]
```diff
@@ -2401,6 +2401,13 @@ public function clear($entityName = null)
...
+                    continue;
```
Comment (Ocramius) [inline lib/Doctrine/ORM/UnitOfWork.php]: In this scenario, using `continue` is more confusing, IMO. `unset()` can live inside the `if()` if you don't negate the comparison

File: lib/Doctrine/ORM/UnitOfWork.php [hunk 3]
```diff
@@ -2401,6 +2401,13 @@ public function clear($entityName = null)
                     $this->doDetach($entity, $visited, false);
                 }
             }
+
+            foreach ($this->entityInsertions as $hash => $entity) {
```
Comment (Ocramius) [inline lib/Doctrine/ORM/UnitOfWork.php]: This block of code (starting at line [2392](https://github.com/doctrine/doctrine2/pull/5850/files#diff-6e8c1c1e78b054ba05e20ea09d877865R2392)) seems to become too complex. Can you move it all into its own private method?

**Comment by Ocramius** [comment]
> Test case provided

Thanks! \o/

File: lib/Doctrine/ORM/UnitOfWork.php (hunk 3 above)
Comment (ricohumme) [inline lib/Doctrine/ORM/UnitOfWork.php]: I could ofcourse, but those are not my contributions. 
Would you like me to alter that ?

File: lib/Doctrine/ORM/UnitOfWork.php (hunk 1 above)
Comment (ricohumme) [inline lib/Doctrine/ORM/UnitOfWork.php]: What do you mean by this?

Comment (Ocramius) [inline lib/Doctrine/ORM/UnitOfWork.php]: `===` or `!==`. Never rely on `==` and `!=` unless strictly required to do so

File: lib/Doctrine/ORM/UnitOfWork.php (hunk 3 above)
Comment (Ocramius) [inline lib/Doctrine/ORM/UnitOfWork.php]: Yes, please :-)

Comment (ricohumme) [inline lib/Doctrine/ORM/UnitOfWork.php]: Fixed in latest commit

File: lib/Doctrine/ORM/UnitOfWork.php (hunk 2 above)
Comment (ricohumme) [inline lib/Doctrine/ORM/UnitOfWork.php]: Fixed in latest commit

File: lib/Doctrine/ORM/UnitOfWork.php (hunk 1 above)
Comment (ricohumme) [inline lib/Doctrine/ORM/UnitOfWork.php]: Fixed in latest commit

Comment (Ocramius) [inline lib/Doctrine/ORM/UnitOfWork.php]: @ricohumme no need to comment on those: they disappear from the diff as soon as you push :-)

**Comment by Ocramius** [comment]
Merged, thanks!

`2.5` - doctrine/doctrine2@0af9ee0140e5ac758e4a1011721810768ceb11cd
`master` - doctrine/doctrine2@3bc61d5f5e4aa75479dfe141e52326243972f530

**Comment by ricohumme** [comment]
Hi,

When will 2.5.5 be released? I'm waiting for this one for one of my projects,

Thnx in advance

**Pull request closed without merging**